warp/Network/Wai/Handler/Warp/Run.hs

```diff
@@ -237,13 +238,18 @@ runServeSettingsConnectionMakerSecure :: Settings
 runServeSettingsConnectionMakerSecure set getConnMaker serveConn = do
     settingsBeforeMainLoop set
     counter <- newCounter
-
-    D.withDateCache $ \dc ->
-        F.withFdCache fdCacheDurationInSeconds $ \fc ->
-            withTimeoutManager $ \tm ->
-                acceptConnection set getConnMaker serveConn dc fc tm counter
+    withII $ acceptConnection set getConnMaker serveConn counter
   where
+    withII action =
+        D.withDateCache $ \dc ->
+        F.withFdCache fdCacheDurationInSeconds $ \fc ->
+        I.withFileInfoCache fdFileInfoDurationInSeconds $ \get ->
+        withTimeoutManager $ \tm -> do
+            let ii0 = InternalInfo undefined tm fc get dc -- fixme: undefined
```

Comment:
I would like to get rid of `undefined`. If we provide `defaultHandle` or something, this can be achieved. However, we need to introduce one more `unsafePerformIO`.

If you think that `undefined` is a necessary evil, let's leave it as it is.